src/app/shared/components/pricing-plans/pricing-plans.component.ts

```diff
@@ -19,6 +19,7 @@ export class PricingPlansComponent implements OnInit, OnDestroy {
   @Input('blockGapsZero') blockGapsZero: boolean; // Flag to add top and bottom gap conditionally
   @Input() showCurrentPlan: boolean;
   @Input() isPrime: boolean;
+  @Input() hideFreePlan: boolean;
```

Comment:
@zainzafar90 you don't need an extra input variable  `hideFreePlan`
because when the `isPrime` is set to `true` that means user is already inside settings,
and if the `isPrime` is set to `false`,  then we have a variable showCurrentPlan, and the current plan now will be free, so we don't show the button on current plan.

Reply:
Okay got it